ocs_ci/utility/psiutils.py

```diff
@@ -0,0 +1,151 @@
+"""
+A module for all PSI-openstack related utilities
+"""
+
+import logging
+
+from keystoneauth1 import loading
+from keystoneauth1 import session
+from cinderclient import client as cinderc
+from novaclient import client as novac
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants, exceptions
+from ocs_ci.utility.utils import TimeoutSampler
+
+logger = logging.getLogger(__name__)
+
+
+class PSIUtils(object):
+    """
+    A class for handling PSI functionalities
+    """
+    def __init__(self, psi_conf):
+        self.auth_url = psi_conf.get('auth_url')
+        self.username = psi_conf.get('username')
+        self.password = psi_conf.get('password')
+        self.project_id = psi_conf.get('project_id')
+        self.user_domain_name = psi_conf.get('user_domain_name')
+
+        self.loader = loading.get_plugin_loader('password')
+        self.auth = self.loader.load_from_options(
+            auth_url=self.auth_url,
+            username=self.username,
+            password=self.password,
+            project_id=self.project_id,
+            user_domain_name=self.user_domain_name
+        )
+        self.sess = session.Session(auth=self.auth)
+        self.nova_clnt = novac.Client(
+            constants.NOVA_CLNT_VERSION, session=self.sess
+        )
+        self.cinder_clnt = cinderc.Client(
+            constants.CINDER_CLNT_VERSION, session=self.sess
+        )
+
+    def create_volume(
+        self, name, size=100, volume_type='tripleo', metadata=None
+    ):
+        """
+        A function to create openstack volumes
+
+        Args:
+            name (str): display name of the volume
+            size (int): of the volume in GBs
+            volume_type (str): type of volume to be created
+            metadata (dict): Any {k,v} to be associated with volume
+
+        Returns:
+            Instance of "Volume" class
```

Comment:
You may have to import but this should show the actual type

```suggestion
            Volume: PSI openstack volume
```

Reply:
ack

Reply:
done